src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -1249,18 +1249,20 @@ public Stream CreateDefaultWin32Resources(bool versionResource, bool noManifest,
 
         internal static void AppendNullResource(Stream resourceStream)
         {
-            var writer = new BinaryWriter(resourceStream);
-            writer.Write((UInt32)0);
-            writer.Write((UInt32)0x20);
-            writer.Write((UInt16)0xFFFF);
-            writer.Write((UInt16)0);
-            writer.Write((UInt16)0xFFFF);
-            writer.Write((UInt16)0);
-            writer.Write((UInt32)0);            //DataVersion
-            writer.Write((UInt16)0);            //MemoryFlags
-            writer.Write((UInt16)0);            //LanguageId
-            writer.Write((UInt32)0);            //Version
-            writer.Write((UInt32)0);            //Characteristics
+            using (var writer = new BinaryWriter(resourceStream, Encoding.UTF8, leaveOpen: true))
```

Comment:
> This ends up calling Flush on the underlying stream, which could have a negative performance impact relative to just letting it flush naturally.

Certainly, this could have that side effect. However, I believe that performance sensitive code should not deviate from recommended coding guideline, unless we have _confirmed_ a performance impact, and then special cased such pieces of code with explicit comments and/or `PerormanceSensitiveAttribute`. At that point, we can revert this change and add a suppression with proper justification.